sys/debug_irq_disable/Kconfig

```diff
@@ -0,0 +1,21 @@
+# Copyright (C) 2022 Benjamin Valentin
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+config MODULE_DEBUG_IRQ_DISABLE
+    bool
+    depends on TEST_KCONFIG
+    depends on CPU_CORE_CORTEX_M
+    help
+        Print time spent with IRQs disabled
+
+config DEBUG_IRQ_DISABLE_THRESHOLD
+    int
+    default 1
+    depends on MODULE_DEBUG_IRQ_DISABLE
+    help
+        Threshold (in CPU ticks) below which periods with IRQs disabled are not printed.
+        Use this to prevent *a lot* of output when debugging.
```

Comment:
Sorry but another suggestion ;-) You have to give the Kconfig entry names. Otherwise it won't show up during `menuconfig`:

```suggestion
menuconfig MODULE_DEBUG_IRQ_DISABLE
    bool "Measure IRQ disable durations"
    depends on TEST_KCONFIG
    depends on CPU_CORE_CORTEX_M
    help
        Print time spent with IRQs disabled

config DEBUG_IRQ_DISABLE_THRESHOLD
    int "Suppress Threshold"
    default 1
    depends on MODULE_DEBUG_IRQ_DISABLE
    help
        Threshold (in CPU ticks) below which periods with IRQs disabled are not printed.
        Use this to prevent *a lot* of output when debugging.
```

Testing module resolution:
```
# Get used modules with Makefile-based reolsution
USEMODULE=debug_irq_disable make -C examples/hello-world/ BOARD=samr30-xpro info-modules >a

# Get used modules with Kconfig
TEST_KCONFIG=1 make -C examples/hello-world/ BOARD=samr30-xpro menuconfig # Enable the module
TEST_KCONFIG=1 make -C examples/hello-world/ BOARD=samr30-xpro info-modules >b

# Check modules lists
diff a b
# No difference :-)
```

Feel free to squash right away.